crates/ruff_linter/src/rules/pyflakes/rules/strings.rs

```diff
@@ -747,7 +747,11 @@ pub(crate) fn string_dot_format_extra_named_arguments(
     let missing: Vec<(usize, &str)> = keywords
         .enumerate()
         .filter_map(|(index, keyword)| {
-            if summary.keywords.contains(keyword.as_ref()) {
+            if summary
```

Comment:
This is unrelated to your change and doesn't need to be fixed as part of this PR: @charliermarsh should we use `HashSet`s instead of `Vec` for these sets (convert from `Vec` to `Set` when converting from `Options` to `Settings`)? Or is the reason we don't do this because we only use `Options` for plugin specific configurations.

Reply:
@MichaReiser -- We could change to `Set`, I don't think there's a specific reason for it. But one problem is that if we _do_ use`HashSet<String>`, then we would need to allocate here to convert from `SmolStr` to `String` for the `contains` check, right?

Reply:
Hmm yeah that might be, because `contains` takes `&T` and not `Deref<T>`. Except if we use `SmolStr` everywhere...

Reply:
That is true... I can try it separately.

Reply:
I should not have brought this up. It's probably not worth it, considering that these vectors only contain very few items and hashing might be more expensive than just comparing the length of every string first.

Reply:
I think that's why I left them as vectors, because the size of these is typically very small.